Dynamic error sampling Python (#8314)
getsentry/sentry-docs: src/platforms/common/configuration/sampling.mdx

</Note>

<PlatformSection supported={["python"]}>

### Dynamically Sampling Error Events

> It is also possible to sample error events dymanically, by setting the <PlatformIdentifier name="error-sampler" /> to a function that returns the desired sample rate for the event. The <PlatformIdentifier name="error-sampler" /> receives two arguments, the <PlatformIdentifier name="event" /> and the <PlatformIdentifier name="hint" />, when called, which it can inspect in order to inform the sampling decision.

Reviewer: ```suggestion
It is also possible to sample error events dynamically, by setting the <PlatformIdentifier name="error-sampler" /> to a function that returns the desired sample rate for the event. The <PlatformIdentifier name="error-sampler" /> receives two arguments, the <PlatformIdentifier name="event" /> and the <PlatformIdentifier name="hint" />, when called, which it can inspect in order to inform the sampling decision.
```

<Alert level="warning">

> The <PlatformIdentifier name="error-sampler" /> function is responsible for returning a valid value. The only valid values are **floating-point numbers** between `0.0` and `1.0`, inclusive (to indicate the probability that an error gets sampled), and **booleans** (to indicate whether or not to sample the error).

Reviewer 2: ```suggestion
Your <PlatformIdentifier name="error-sampler" /> function must return a valid value. A valid value is a **floating-point number** between `0.0` and `1.0` (inclusive) indicating the probability an error gets sampled) or a **boolean** indicating whether or not to sample the error.
```
